src/StatusesHandler.php

```diff
@@ -0,0 +1,270 @@
+<?php
+
+namespace RebelCode\Bookings\WordPress\Module;
+
+use Dhii\Invocation\InvocableInterface;
+use Dhii\Util\Normalization\NormalizeArrayCapableTrait;
+use Psr\EventManager\EventManagerInterface;
+use RebelCode\Modular\Events\EventsConsumerTrait;
+use Psr\EventManager\EventInterface;
+use Dhii\Event\EventFactoryInterface;
+use Traversable;
+
+/**
+ * Handler for providing the status related state and functionality.
+ *
+ * @since [*next-version*]
+ */
+class StatusesHandler implements InvocableInterface
+{
+    /* @since [*next-version*] */
+    use EventsConsumerTrait;
+
+    /* @since [*next-version*] */
+    use NormalizeArrayCapableTrait;
+
+    /**
+     * List of statuses keys available in application.
+     *
+     * @since [*next-version*]
+     *
+     * @var array
+     */
+    protected $statuses;
+
+    /**
+     * List of statuses keys that are not visible in the UI.
+     *
+     * @since [*next-version*]
+     *
+     * @var array
+     */
+    protected $hiddenStatuses;
+
+    /**
+     * Map of known statuses keys translations.
+     *
+     * @since [*next-version*]
+     *
+     * @var array
+     */
+    protected $statusesLabels;
+
+    /**
+     * Option key name to save screen statuses config.
+     *
+     * @since [*next-version*]
+     *
+     * @var string
+     */
+    protected $screenOptionsKey;
+
+    /**
+     * Endpoint for saving statuses.
+     *
+     * @since [*next-version*]
+     *
+     * @var string
+     */
+    protected $statusesEndpoint;
+
+    /**
+     * StatusesHandler constructor.
+     *
+     * @since [*next-version*]
+     *
+     * @param Traversable           $statuses         List of statuses key in application.
+     * @param Traversable           $hiddenStatuses   List of hidden statuses in application.
+     * @param Traversable           $statusesLabels   Map of known statuses labels.
+     * @param string                $screenOptionsKey Option key name to save screen statuses config.
+     * @param string                $statusesEndpoint Endpoint for saving statuses.
+     * @param EventManagerInterface $eventManager     The event manager.
+     * @param EventFactoryInterface $eventFactory     The event factory.
+     */
+    public function __construct(
+        Traversable $statuses,
+        Traversable $hiddenStatuses,
+        Traversable $statusesLabels,
+        $screenOptionsKey,
+        $statusesEndpoint,
+        $eventManager,
+        $eventFactory
+    ) {
+        $this->statuses       = $this->_normalizeArray($statuses);
```

Comment:
I don't see any usage of statuses here. Why the normalization?

Reply:
I decided to do normalization here because in other 2 methods I need normalize this statuses. So I decided to store it as an array and work with array.